src/contributing/release_process.md

```diff
@@ -0,0 +1,164 @@
+expectations would include information about supported operating system versions or distributions,
+supported hardware or hardware configurations, or non-functional expectations such as performance or
+scalability. Parsec is a public open source project and it can be used in a variety of contexts. It
+is the responsibility of the system integrator or Parsec adopter to ensure its suitability for use
+in any product or other context. Any questions around specific expectations and requirements can be
+raised with the Parsec maintainers through the available [community
+channels](https://github.com/parallaxsecond/community).
+
+## Feature Branches
+
+For complex features whose development is likely to go beyond the code freeze date for the next
+scheduled release, the use of feature branches may be appropriate, and this can be agreed between
+the relevant contributors and project maintainers on a case-by-case basis. It is not always
+necessary to use a feature branch. However, if any partially-complete features are merged into the
+main branch, this *must* be done in a way that does not disrupt the release process. To ensure this,
+please abide by the following rules:
+
+- Use [Cargo features](https://doc.rust-lang.org/cargo/reference/features.html) to allow the changes
+   to be compiled conditionally, and ensure that any new dependencies are made optional and
+   conditional on the Cargo features.
+- Avoid the use of [Git
+   dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories)
+   in the main branch of Parsec repositories. Git dependencies are sometimes appropriate and useful
+   in feature branches, but they should be avoided in main branches where possible. Git dependencies
+   block the release process, and they must always be replaced with numbered crate versions well
+   before the next code freeze period.
+- Ensure that all relevant Continuous Integration tests are passing for code that is promoted to
+   main branches.
+
+If you are planning to contribute a significant feature or major architectural change, please
+contact the maintainers via the available community channels, such as the Slack channel or the
+weekly meetings. Please see the [community](https://github.com/parallaxsecond/community) repository
+for more details. The project maintainers will be available to advise you on whether to use a
+feature branch.
+
+When using feature branches, please ensure that commits are reviewed by project maintainers at the
+time of submission to the feature branch. Do not wait until the entire feature is ready to be
+promoted into the main branch before requesting code reviews, since large and bulky code reviews are
+generally not well received by the maintainers, and this might delay the promotion of the changes.
+
+To keep the repositories tidy, please delete feature branches or other temporary branches as soon as
+they are no longer needed.
```

Comment:
I think this is something only the maintainers can do, but the creator of a feature branch could inform the maintainer that it is ready to be deleted.

Reply:
Yes, good point - will reword this,